### ISSUE TITLE
XML NodeSet Compiler: AttributeError, no attribute "stringRepresentation"

### ISSUE THREAD

**Issue by NIPE-SYSTEMS** [open62541/open62541#1823]
Description
===========

I'm trying to compile a XML nodeset to C code with the nodeset compiler. After making changes to the information model XML nodeset, the compilation fails (see diff below).

Background Information / Reproduction Steps
===========================================

While looking at the sourcecode of [tools/nodeset_compiler/datatypes.py](https://github.com/open62541/open62541/blob/4d8f829550995ac8d7358fa69c7d9dfc69e96326/tools/nodeset_compiler/datatypes.py) it seems odd that `self.stringRepresentation` never gets defined in the `NodeId`-class nor the `Value`-class. But here it gets used which results in an `AttributeError`:

https://github.com/open62541/open62541/blob/4d8f829550995ac8d7358fa69c7d9dfc69e96326/tools/nodeset_compiler/datatypes.py#L562-L569

Log output of the compilation process
------------------------------------------

```
INFO:__main__:Generating Code
INFO:__main__:Preprocessing ../../../nodeset/example.xml
Traceback (most recent call last):
  File "nodeset_compiler.py", line 179, in <module>
    ns.allocateVariables()
  File "/ba/open62541/tools/nodeset_compiler/nodeset.py", line 277, in allocateVariables
    n.allocateValue(self)
  File "/ba/open62541/tools/nodeset_compiler/nodes.py", line 294, in allocateValue
    self.value.parseXMLEncoding(self.xmlValueDef, dataTypeNode, self)
  File "/ba/open62541/tools/nodeset_compiler/datatypes.py", line 166, in parseXMLEncoding
    self.value = [self.__parseXMLSingleValue(xmlvalue, parentDataTypeNode, parent)]
  File "/ba/open62541/tools/nodeset_compiler/datatypes.py", line 204, in __parseXMLSingleValue
    t.parseXML(xmlvalue)
  File "/ba/open62541/tools/nodeset_compiler/datatypes.py", line 567, in parseXML
    if not self.stringRepresentation == xmlvalue.localName:
AttributeError: 'NodeId' object has no attribute 'stringRepresentation'
CMakeFiles/example-nodeset.dir/build.make:60: recipe for target 'src_generated/example.c' failed
make[2]: *** [src_generated/example.c] Error 1
CMakeFiles/Makefile2:106: recipe for target 'CMakeFiles/example-nodeset.dir/all' failed
make[1]: *** [CMakeFiles/example-nodeset.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
INFO:__main__:NodeSet generation code successfully printed
```

Change of the XML nodeset that results in the error
-------------------------------------------------------------------------------

```diff
diff --git a/nodeset/example.xml b/nodeset/example.xml
index f2fb5c9..0ccf2f4 100644
--- a/nodeset/example.xml
+++ b/nodeset/example.xml
@@ -4,8 +4,10 @@
         <Uri>http://example.de/UA/</Uri>
     </NamespaceUris>
     <Aliases>
+        <Alias Alias="Boolean">i=1</Alias>
         <Alias Alias="String">i=12</Alias>
         <Alias Alias="ByteString">i=15</Alias>
+        <Alias Alias="NodeId">i=17</Alias>
         <Alias Alias="LocalizedText">i=21</Alias>
         <Alias Alias="HasModellingRule">i=37</Alias>
         <Alias Alias="HasTypeDefinition">i=40</Alias>
@@ -152,6 +154,42 @@
             <uax:Int32>0</uax:Int32>
         </Value>
     </UAVariable>
+    <UAObjectType NodeId="ns=1;i=1005" BrowseName="1:ElementsChangeEventType">
+        <DisplayName Locale="en">ElementsChangeEventType</DisplayName>
+        <Description Locale="en">Event type that notifies about a change of the list of elements.</Description>
+        <References>
+            <Reference ReferenceType="HasSubtype" IsForward="false">i=2041</Reference>
+            <Reference ReferenceType="HasComponent">ns=1;i=6009</Reference>
+            <Reference ReferenceType="HasComponent">ns=1;i=6011</Reference>
+            <Reference ReferenceType="GeneratesEvent" IsForward="false">ns=1;i=5001</Reference>
+        </References>
+    </UAObjectType>
+    <UAVariable DataType="NodeId" ParentNodeId="ns=1;i=1005" NodeId="ns=1;i=6009" BrowseName="1:Element" UserAccessLevel="3" AccessLevel="3">
+        <DisplayName Locale="en">Element</DisplayName>
+        <Description Locale="en">The element that has been added/removed.</Description>
+        <References>
+            <Reference ReferenceType="HasTypeDefinition">i=63</Reference>
+            <Reference ReferenceType="HasModellingRule">i=78</Reference>
+            <Reference ReferenceType="HasComponent" IsForward="false">ns=1;i=1005</Reference>
+        </References>
+        <Value>
+            <uax:NodeId>
+                <uax:Identifier>i=0</uax:Identifier>
+            </uax:NodeId>
+        </Value>
+    </UAVariable>
+    <UAVariable DataType="Boolean" ParentNodeId="ns=1;i=1005" NodeId="ns=1;i=6011" BrowseName="1:IsAddition" UserAccessLevel="3" AccessLevel="3">
+        <DisplayName Locale="en">IsAddition</DisplayName>
+        <Description Locale="en">True if this event corresponds to an addition, false if it corresponds to a removal.</Description>
+        <References>
+            <Reference ReferenceType="HasTypeDefinition">i=63</Reference>
+            <Reference ReferenceType="HasModellingRule">i=78</Reference>
+            <Reference ReferenceType="HasComponent" IsForward="false">ns=1;i=1005</Reference>
+        </References>
+        <Value>
+            <uax:Boolean>false</uax:Boolean>
+        </Value>
+    </UAVariable>
     <UAObjectType NodeId="ns=1;i=1013" BrowseName="1:StorageType">
         <DisplayName Locale="en">StorageType</DisplayName>
         <Description Locale="en">StorageType</Description>
@@ -216,7 +254,8 @@
         <DisplayName Locale="en">Elements</DisplayName>
         <Description Locale="en">Elements</Description>
         <References>
+            <Reference ReferenceType="GeneratesEvent">ns=1;i=1005</Reference>
             <Reference ReferenceType="HasModellingRule">i=78</Reference>
             <Reference ReferenceType="HasComponent" IsForward="false">ns=1;i=1013</Reference>
         </References>
```

The node likely producing the error
----------------------------------------

```xml
<UAVariable DataType="NodeId" ParentNodeId="ns=1;i=1005" NodeId="ns=1;i=6009" BrowseName="1:Element" UserAccessLevel="3" AccessLevel="3">
    <DisplayName Locale="en">Element</DisplayName>
    <Description Locale="en">The element that has been added/removed.</Description>
    <References>
        <Reference ReferenceType="HasTypeDefinition">i=63</Reference>
        <Reference ReferenceType="HasModellingRule">i=78</Reference>
        <Reference ReferenceType="HasComponent" IsForward="false">ns=1;i=1005</Reference>
    </References>
    <Value>
        <uax:NodeId>
            <uax:Identifier>i=0</uax:Identifier>
        </uax:NodeId>
    </Value>
</UAVariable>
```

Checklist
=========

 - [x] open62541 Version (release number or git tag): 4d8f829550995ac8d7358fa69c7d9dfc69e96326
 - [x] ~~Other OPC UA SDKs used (client or server):~~ *Not applicable*
 - [x] Operating system: *Ubuntu 18.04 Docker container running on Arch Linux* (`FROM ubuntu:18.04`)
 - [x] Logs (with `UA_LOGLEVEL` set as low as necessary) attached: *see above*
 - [x] ~~Wireshark network dump attached~~ *Not applicable*
 - [x] Self-contained code example attached: *see above*
 - [x] ~~Critical issue~~ *No, I have a workaround (I'm using Integer as datatype until this issue is fixed)*

**Comment by jpfr** [comment]
Thanks for reporting.
The fix is coming in #1828.

**Comment by NIPE-SYSTEMS** [comment]
Thanks!
